test/EFCore.PG.FunctionalTests/MonsterFixupChangedChangingNpgsqlTest.cs

```diff
@@ -17,34 +16,14 @@ public class MonsterFixupChangedChangingNpgsqlFixture : MonsterFixupChangedChang
         {
             protected override ITestStoreFactory TestStoreFactory => NpgsqlTestStoreFactory.Instance;
 
-            public override DbContextOptionsBuilder AddOptions(DbContextOptionsBuilder builder)
-                => base.AddOptions(builder).ConfigureWarnings(w => w.Log(RelationalEventId.QueryClientEvaluationWarning));
-
             protected override void OnModelCreating<TMessage, TProduct, TProductPhoto, TProductReview, TComputerDetail, TDimensions>(
                 ModelBuilder builder)
             {
                 base.OnModelCreating<TMessage, TProduct, TProductPhoto, TProductReview, TComputerDetail, TDimensions>(builder);
 
-                builder.Entity<TMessage>().Property(e => e.MessageId).UseNpgsqlSerialColumn();
-
-                builder.Entity<TProduct>()
-                    .OwnsOne(c => (TDimensions)c.Dimensions, db =>
-                    {
-                        db.Property(d => d.Depth).HasColumnType("decimal(18,2)");
-                        db.Property(d => d.Width).HasColumnType("decimal(18,2)");
-                        db.Property(d => d.Height).HasColumnType("decimal(18,2)");
-                    });
-
-                builder.Entity<TProductPhoto>().Property(e => e.PhotoId).UseNpgsqlSerialColumn();
-                builder.Entity<TProductReview>().Property(e => e.ReviewId).UseNpgsqlSerialColumn();
-
-                builder.Entity<TComputerDetail>()
-                    .OwnsOne(c => (TDimensions)c.Dimensions, db =>
-                    {
-                        db.Property(d => d.Depth).HasColumnType("decimal(18,2)");
-                        db.Property(d => d.Width).HasColumnType("decimal(18,2)");
-                        db.Property(d => d.Height).HasColumnType("decimal(18,2)");
-                    });
+                builder.Entity<TMessage>().Property(e => e.MessageId).ForNpgsqlUseSerialColumn();
+                builder.Entity<TProductPhoto>().Property(e => e.PhotoId).ForNpgsqlUseSerialColumn();
+                builder.Entity<TProductReview>().Property(e => e.ReviewId).ForNpgsqlUseSerialColumn();
```

Comment:
What about?
```c#
if (TestEnvironment.PostgresVersion < new Version(10, 0)
{
    builder.Entity<TMessage>().Property(e => e.MessageId).ForNpgsqlUseSerialColumn();
    builder.Entity<TProductPhoto>().Property(e => e.PhotoId).ForNpgsqlUseSerialColumn();
    builder.Entity<TProductReview>().Property(e => e.ReviewId).ForNpgsqlUseSerialColumn();
}
else
{
    builder.Entity<TMessage>().Property(e => e.MessageId).ForNpgsqlUseIdentityColumn();
    builder.Entity<TProductPhoto>().Property(e => e.PhotoId).ForNpgsqlUseIdentityColumn();
    builder.Entity<TProductReview>().Property(e => e.ReviewId).ForNpgsqlUseIdentityColumn();
}

Reply:
It's possible, but that test doesn't care at all about what kind of value-generated column it uses - serial is just as good as identity. And as serial is out current default it makes sense to simply use that for now...

We can revisit once we switch the default to identity.